types/ui-grid/index.d.ts

```diff
@@ -3799,7 +3799,7 @@ declare namespace uiGrid {
          * If suppressRemoveSort is also set, the unsorted state will be skipped even if it is listed here. Each direction may
          * not appear in the list more than once (e.g. [ASC, DESC, DESC] is not allowed), and the list may not be empty.*
          */
-        sortDirectionCycle?: Array<IUiGridConstants>;
+        sortDirectionCycle?: Array<null | string>;
```

Comment:
Granted, you would also need to change https://github.com/hadalin/DefinitelyTyped/blob/330059eccdda88277b92bb4e31f9077f4279e2d3/types/ui-grid/index.d.ts#L87 :

```ts
   ASC: 'asc',
   DESC: 'desc',
```